### PR TITLE
Get rid of through2 lib usage

### PR DESCRIPTION
**Pull request by rozap**
* through2 hardcodes the highwatermark
  which is problematic for complex shapes
* ldjson stream uses through2, so drop that
  as well
* re-implement ldjson stream in ldjson.js
* re-implement through2 reprojection that lived
  in layer.js in wgs84-reprojector

### CODE REVIEW

File: es6-lib/decoders/ldjson.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      highWaterMark: config().rowBufferSize
+    });
+    this._buf = '';
+  }
+
+  _push(token) {
+    try {
+      this.push(JSON.parse(token));
+    } catch(e) {
+
```
Comment (catstavi) [inline es6-lib/decoders/ldjson.js]: Do nothing here?

Comment (rozap) [inline es6-lib/decoders/ldjson.js]: this is a `''` token

i suppose there could be a warning if it fails on something other than `''`

**Comment by catstavi** [comment]
Okay, I think this looks good. The new classes should probably have tests, though?

**Comment by rozap** [comment]
yea it would be good to test them at a finer level of granularity. i just need to get this merged now so we can run the regression suite against them. important to note that these are tested in the `unit/merger` test suite as well as the smoke tests. but i'll add more asap, i've opened #34 to track this

**Pull request merged**